pandas_datareader/oandarest.py

```diff
@@ -0,0 +1,303 @@
+                "granularity": granularity,
+                "from": current_start_rfc3339,
+                "to": current_end_rfc3339,
+                "price": candleFormat,
+                "smooth": "false",
+                "includeFirst": "true" if includeCandleOnStart else "false",
+                "dailyAlignment": "17",
+                "alignmentTimezone": "America/New_York",
+                "weeklyAlignment": "Friday"
+            }
+
+            # print(params)
+
+            try:
+                request = instruments.InstrumentsCandles(
+                    instrument=currencyPair,
+                    params=params)
+
+                response = oanda.request(request)
+                current_start = current_end
+                includeCandleOnStart = False
+            except Exception as error:
+                isExceedResultsLimitError = re.findall(
+                    "The results returned satisfying the query exceeds the maximum size", str(error))
+                isExceedResultsLimitError = True if len(
+                    isExceedResultsLimitError) else False
+
+                if isExceedResultsLimitError:
+                    # Problem: OANDA caps returned range to 5000 results max
+                    # Solution: Reduce requested date interval to return less
+                    # than 5000 results
+                    current_duration /= 2
+                    continue
+                else:
+                    print("ERROR OANDA: " + str(error))
+                    print(str(error))
+                    raise error
+
+            # print(response)
+
+            if not response:
+                continue
+
+            candles = response['candles']
+
+            if not candles:
+                continue
+
+            # print(candles)
+
+            ohlc = ['o', 'h', 'l', 'c']
+            df = pd.io.json.json_normalize(
+                candles,
+                meta=['time', 'volume', 'complete', [
+                    'mid', ohlc], ['ask', ohlc], ['bid', ohlc]]
+            )
+            df['time'] = pd.to_datetime(df['time'])
+
+            with pd.option_context('display.max_columns', None):
+                pass  # print(df)
+
+            dfs.append(df)
+
+        df = pd.concat(dfs)
+
+        df.rename(inplace=True,
+                  index=str,
+                  columns={
+                      "time": "date",
+                      "mid.o": "mid.open",
+                      "mid.h": "mid.high",
+                      "mid.l": "mid.low",
+                      "mid.c": "mid.close",
+                      "ask.o": "ask.open",
+                      "ask.h": "ask.high",
+                      "ask.l": "ask.low",
+                      "ask.c": "ask.close",
+                      "bid.o": "bid.open",
+                      "bid.h": "bid.high",
+                      "bid.l": "bid.low",
+                      "bid.c": "bid.close",
+                  })
+
+        df = df.set_index('date')
```

Comment:
Fixed: Use `Open` instead `open`, etc ...

I will look at A DataFrame with a MultiIndex later this week.